esphome/components/esp32_can/canbus.py

```
@@ -0,0 +1,29 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import pins
+from esphome.components import canbus
+from esphome.const import CONF_ID, CONF_RX_PIN, CONF_TX_PIN, ESP_PLATFORM_ESP32
+from esphome.components.canbus import CanbusComponent
+
+esp32_can_ns = cg.esphome_ns.namespace("esp32_can")
+esp32_can = esp32_can_ns.class_("ESP32Can", CanbusComponent)
+
+ESP_PLATFORMS = [ESP_PLATFORM_ESP32]
+
+CONFIG_SCHEMA = canbus.CANBUS_SCHEMA.extend(
+    {
+        cv.GenerateID(): cv.declare_id(esp32_can),
+        cv.Required(CONF_RX_PIN): pins.gpio_output_pin_schema,
+        cv.Required(CONF_TX_PIN): pins.gpio_output_pin_schema,
+    }
+)
```

There are some speeds valid in config which are not valid in the cpp, for these the code will just fail silently. 

You can add here a check on the configuration to see if the config speed is not one of the supported in your cpp. 

Look for cv.All( ... , validate_xxxx) 

You must create this validate function

---

I couldn't quite figure out how this worked. Therefore I just overrode the possible bit rate values with a subset.